### PR TITLE
Update Sorbet and fix failing test case on `main`

### PR DESCRIPTION
**Pull request by andyw8**
### Motivation

Due to https://github.com/sorbet/sorbet/pull/6426, the Sorbet output changed, causing a test failure in `main`.

### Implementation

* Ran `bundle update sorbet`
* Adjusted the test based on the output from Sorbet, with the assumption that the output is correct.
* Ran `bundle exec exe/tapioca gem` and committed the RBI changes.

### Tests

n/a

### CODE REVIEW

File: sorbet/rbi/gems/mini_portile2@2.8.0.rbi [hunk 1]
```diff
@@ -1,8 +0,0 @@
-# typed: true
```
Comment (andyw8) [inline sorbet/rbi/gems/mini_portile2@2.8.0.rbi:1]: Removed by `bundle exec exe/tapioca gem`.

**Comment by andyw8** [comment]
@paracycle I saw that you added Ruby 3.2 to the build in https://github.com/Shopify/tapioca/pull/1214 as 'experimental'. If a Ruby 3.2 job fails, is the intention that the 'overall' build should still pass?

**Comment by andyw8** [comment]
Resolved by #1227.

**Comment by paracycle** [comment]
> @paracycle I saw that you added Ruby 3.2 to the build in #1214 as 'experimental'. If a Ruby 3.2 job fails, is the intention that the 'overall' build should still pass?

I see this is closed, but yes, that's the general idea. The Ruby 3.2 job is to test Ruby 3.2 more than Tapioca itself. For example, last year it had caught this: https://bugs.ruby-lang.org/issues/18405

**Pull request closed without merging**